Core/Base/AgentModule.py

```diff
@@ -337,6 +347,11 @@ def am_go( self ):
       cD = self.__moduleProperties[ 'cyclesDone' ]
       self.log.notice( "Remaining %s of %s cycles" % ( mD - cD, mD ) )
     self.log.notice( "-"*40 )
+    # use SIGALARM as a watchdog interrupt if enabled
+    watchdogInt = self.am_getWatchdogTime()
+    if watchdogInt > 0:
+      signal.signal( signal.SIGALRM, signal.SIG_DFL )
```

Comment:
Just in-case something has already set-up a SIGALRM handler at some point, this makes sure we get the expected behaviour rather than triggering a random bit of code. It isn't strictly necessary, but is there to try and minimise surprises in the future.